documentation/decisions/adr/2023-12-18-measurement-dashboard-architecture.md

```diff
@@ -0,0 +1,149 @@
+
+### Positive Consequences <!-- OPTIONAL -->
+
+- {e.g., improved performance on quality metric, new capability enabled, ...}
+- ...
+
+### Negative Consequences <!-- OPTIONAL -->
+
+- {e.g., decreased performance on quality metric, risk, follow-up decisions required, ...}
+- ...
+
+## Pros and Cons of the Options <!-- OPTIONAL -->
+
+### Analytics API + user interface
+
+This option involves building both a custom analytics API that will serve the data behind our key project metrics and a separate user interface to consume from that API and render these metrics as a dashboard for end users. Following this approach, we could choose to re-use our existing front-end and backend infrastructure or make slightly different tooling choices based on our needs.
+
+> [!TIP]
+> **Bottom line:** This option is best if we want to follow the existing architecture pattern of Simpler.Grants.gov and have maximum control over the structure and design of the public dashboard, but can dedicate more engineering resources to building and hosting this dashboard and don't need to update the analytics endpoints or dashboards frequently.
+
+- **Pros**
+  - Provides the most fine-grained control over the look and feel of the dashboard application.
+  - Enables us to combine multiple dashboards into a single application.
+  - Enables S2S users to access underlying analytics data via API.
+  - Allows us to include narrative text that explains the metrics we publish.
+  - Enables open source contributors to host local versions of our dashboard or easily create their own dashboards.
+- **Cons**
+  - Building or modifying dashboards is quite technical, not something a business analyst could do on their own.
+  - Publishing new charts or dashboards requires making changes both to the API and to the dashboard UI separately.
+  - Hardest to maintain of all of the options considered.
+  - Does not easily support adhoc reports or dashboards without using a separate tool.
+
+### Custom dashboard app
+
+This option involves building a custom dashboard app using an existing framework, such as [Dash][plotly-dash] (python) or [R Shiny][r-shiny]. This custom built application would be hosted as its own application and be slightly more integrated than a separate analytics API + dashboard UI, but it would be more customizable than using an open source or proprietary dashboard solution.
+
+> [!TIP]
+> **Bottom line:** This option would be best if we want to retain control over the look and feel of the dashboard and manage the backend and frontend of the dashboard together, but can dedicate more engineering resources to building and hosting it and don't need to expose analytics data via API.
+
+- **Pros**
+  - Provides more fine-grained control over the look and feel of the dashboard application, on par with option 1.
+  - Allows a single engineer familiar with Python (or R if we use Shiny) to manage both the backend (e.g. analytics) and frontend (visualization and design of the dashboard).
```

Comment:
I have trigger warnings on Shiny apps 😅 . R is a very powerful and convoluted language, haha.

Reply:
Not that it would be the best option, but Shiny [https://shiny.posit.co/py/](now works with Python)